numpy/f2py/capi_maps.py

```diff
@@ -126,13 +126,17 @@
               'byte': {'': 'char'},
               }
 
-f2cmap_all = deep_merge(f2cmap_all, iso_c_binding_map)
+# Add ISO_C handling
+c2pycode_map.update(isoc_c2pycode_map)
+c2py_map.update(iso_c2py_map)
+f2cmap_all, _ = process_f2cmap_dict(f2cmap_all, iso_c_binding_map, c2py_map)
+# End ISO_C handling
 f2cmap_default = copy.deepcopy(f2cmap_all)
```

Comment:
The problem with the old handling is that updating the `f2cmap_all` doesn't preserve / update the mapping basically.